### DISCUSSION TITLE
Can we exec a upsert?

### DISCUSSION THREAD

**Discussion by niuhuan** [SeaQL/sea-orm#650] [General]
I need the upsert function. I'm not found it form [docs](https://www.sea-ql.org/SeaORM/docs/basic-crud/insert) and issues.
I hope this operation will be implemented in SeaOrm. This can reduce error of writing raw sql.
 I am a novice with rust. These codes are just my imagination.

```rust
let pear = fruit::ActiveModel {
    name: Set("Pear".to_owned()),     // update column if row exists
    nickname: SetOrKeep("pear chan"),     // not modify if row exists
    f_count: SetOrIncrease(1),     // Increase if row exists
    ..Default::default()
};
```

Their effect should be like this in the database

```mysql
       
INSERT INTO fruit(id, name, nickname, f_count)
        VALUE ("Pear","pear chan",1) ON DUPLICATE KEY UPDATE name="Pear", f_count = f_count+1;

```

**Comment by RomainMazB** [comment]
The upsert function is not available yet, there is an active PR in sea-query that needs to be merged before this append in sea-orm.

Mentionned in:
- https://github.com/SeaQL/sea-orm/issues/187
- https://github.com/SeaQL/sea-query/pull/167

**Reply by kudlatyamroth** [reply]
as Sebastien said here: https://github.com/SeaQL/sea-orm/issues/187#issuecomment-1089810692
on conflict is merged in sea orm in 0.24 :)
